js/ui/grid_core/ui.grid_core.data_controller.js

```diff
@@ -492,6 +492,11 @@ module.exports = {
 
                     that.callBase();
                     dataSource = that._dataSource;
+
+                    if(dataSource && that.option("keyExpr") && dataSource.key() !== that.option("keyExpr")) {
```

Comment:
Maybe better replace it with a single expression that tells what do you check? What do you think?

Reply:
I'll do that. The code will become more readable.